opentech/apply/users/views.py

```diff
@@ -12,3 +15,13 @@ def account(request):
     return render(request, 'users/account.html', {
         'show_change_password': password_management_enabled() and request.user.has_usable_password(),
     })
+
+
+@login_required(login_url=reverse_lazy('users:login'))
+@require_oauth_whitelist
```

Comment:
You duplicate the logic for the oauth white listing in the decorator and in the tags. I would prefer to just use the version you have in the tags (moved to utils?) and combine that with the view decorator `user_passes_test`

https://docs.djangoproject.com/en/2.0/topics/auth/default/#django.contrib.auth.decorators.user_passes_test

Reply:
👍 
was thinking about that.

Reply:
actually, cannot use `user_passes_test` as this is two-fold:
1. needs to be logged-in, hence `@login_required`
2. logged-in users that cannot use OAuth should see a 403, not a login page

Reply:
Valid points. I think the changes you have made alleviate most of my concerns.

`login_required` is already pointing at settings.LOGIN_URL (`'users:login'`)